speaking-portal-project/src/animationFactory/animationProcessor.ts

```diff
@@ -0,0 +1,155 @@
+import { MouthCue, MouthCueArray } from '../types'
+import fs from 'fs'
+
+export async function mouthCuesToInputFile({
+    characterSelect,
+    mouthCues,
+    outputPath = './tmp/input.txt',
+}: {
+    characterSelect: string
+    mouthCues: MouthCue[]
+    outputPath?: string
+}) {
+    try {
+        await fs.promises.writeFile(outputPath, generateFrameData(characterSelect, mouthCues), {
+            flag: 'w',
+        })
+        /* Lines 17-20 for testing purposes only
+        const contents = await fs.promises.readFile(outputPath, 'utf-8')
+        console.log(contents);
+
+        return contents
+        */
+    } catch (err) {
+        console.log(err)
+        return Error('MouthCuesFileError')
+    }
+}
+
+/*
+This converts  data from the MouthCue type into a format for ffmpeg
+*/
+export function generateFrameData(avatar: string, mouthCues: MouthCue[]) {
+    let frameData = ''
+    let blink = false
+    let lastBlink = -1
+    let lastPoseChange = -1
+    let idle = true
+    let breathPhase = 0 // 0 = rest, 1 = inhaling, 2 = lungs full, 3 = exhaling
+
+    console.log("Adding realism...")
+    for (const mouthCuesKey in mouthCues) {
+        // Duration of frame is calculated based on the mouth cue
+        let frameDur = (mouthCues[mouthCuesKey].end - mouthCues[mouthCuesKey].start).toFixed(2)
+        let currentSec = parseInt(mouthCues[mouthCuesKey].start.toFixed(0))
+
+        // TODO: use for testing. After confirming that blinking and other realism is good, delete this
+        console.log("t=%d, frameDur: %d, lastBlink: %d, idling? %s, lastPoseChange: %d",
+                    currentSec, frameDur, lastBlink, lastPoseChange, idle, lastPoseChange)
+
+        // Start building frame's path
+        frameData.concat(`file ../images/${avatar}/${mouthCues[mouthCuesKey].value}`)
+
+        // Blinking
+        blink = willBlink(currentSec, frameDur, lastBlink)
+        if (blink) {
+            lastBlink += 1
+            frameData.concat('_blink')
+        }
+
+        // TODO: determine if avatar will do a body pose or be in idle animation
+
+        // Idling or Posing
+        if(idle){
+            breathPhase = calculateBreathPhase(currentSec, lastPoseChange, breathPhase)
+            frameData.concat(translateBreathFrame(breathPhase))
+        } else {
+            breathPhase = 0
+        }
+
```

Comment:
I think we will eventually need to add additional checks here, for example if the character is not idle, what if it's raising its hand, or doing a little dance, etc?

Reply:
Oh definitely, I forgot to add the todo here! Good catch